### PR TITLE
Sample BetaGeo population parameters

### PR DESCRIPTION
**Pull request by wd60622**
Addresses the [issue here](https://github.com/pymc-labs/pymc-marketing/issues/301)

~Currently only address models fit with mcmc~

----
:books: Documentation preview :books:: https://pymc-marketing--303.org.readthedocs.build/en/303/

### CODE REVIEW

**Comment by wd60622** [comment]
Fixes the docs [here](https://pymc-marketing--303.org.readthedocs.build/en/303/api/generated/pymc_marketing.clv.models.beta_geo.BetaGeoModel.html#pymc_marketing.clv.models.beta_geo.BetaGeoModel)

Docs for the new method are [here](https://pymc-marketing--303.org.readthedocs.build/en/303/api/generated/pymc_marketing.clv.models.beta_geo.BetaGeoModel.sample_population_distributions.html#pymc_marketing.clv.models.beta_geo.BetaGeoModel.sample_population_distributions)

**Comment by wd60622** [comment]
@ricardoV94

File: pymc_marketing/clv/models/beta_geo.py
Comment (ricardoV94) [inline pymc_marketing/clv/models/beta_geo.py:350]: Maybe keep the name of the methods more similar to those in the gamma-gamma? Something like `distribution_new_customer_dropout` and alike. Internally they can use this method which should be pretty cheap anyway.

I think that will make them more discoverable

File: tests/clv/models/test_beta_geo.py
Comment (ricardoV94) [inline tests/clv/models/test_beta_geo.py:340]: Can we assert something more concrete, like the mean/std is close enough to what would be expected from the fit?

Also let's not do an actual fit which would be a relatively slow test. You can borrow the logic in the other tests of doing a fake fit from prior_predictive

File: pymc_marketing/clv/models/beta_geo.py
Comment (ColtAllen) [reply pymc_marketing/clv/models/beta_geo.py:350]: Do you think this method could be moved into `BaseModel` and standardized across all models?

File: tests/clv/models/test_beta_geo.py
Comment (wd60622) [reply tests/clv/models/test_beta_geo.py:340]: Yes. Good idea

File: pymc_marketing/clv/models/beta_geo.py
Comment (wd60622) [reply pymc_marketing/clv/models/beta_geo.py:350]: Good thinking. 
I view this as the closest solution

https://github.com/pymc-labs/pymc-marketing/blob/d4feeeafad6f0299b115dac3582ad6d98bd8241f/pymc_marketing/clv/models/shifted_beta_geo.py#L158-L201

**Comment by wd60622** [comment]
> Looks good so far. Only suggestion I have is to add a few more sentences to the docstrings explaining how to interpret the distributions. The Gamma distribution is simple as it describes the time between transactions for the customer population, but the Beta is a little trickier:
> 
> It illustrates the probability of a customer permanently dropping out after making a repeat purchase. Transactions for the BG/NBD model are assumed independent and identically distributed, and if a customer has only made a single purchase, then they are assumed 100% still active.

Yes, I will add to the docstrings. Good suggestion. 

I will just use the verbiage from [Haddie's paper](http://www.brucehardie.com/papers/bgnbd_2004-04-20.pdf) because, imo, the inclusion or exclusion of a "zero point" in the data will require the "repeat". 
Let me know if you have a different take

**Comment by wd60622** [comment]
Method summary
Screenshot 2023-06-25 at 17 04 24

distribution_new_customer_dropout
Screenshot 2023-06-25 at 17 04 55

distribution_new_customer_purchase_rate

Screenshot 2023-06-25 at 17 05 09

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Looks great @wd60622, we might just wait for the large refactor PR of the CLV models to be merged first before updating this and merging it. Easier to solve conflicts here than there.

**Comment by wd60622** [comment]
> Looks great @wd60622, we might just wait for the large refactor PR of the CLV models to be merged first before updating this and merging it. Easier to solve conflicts here than there.

Perfect. Sounds good. Thanks for the feedback

**Comment by ricardoV94** [comment]
@wd60622 we merged the big CLV refactor PR #298. Do you want to update this one so we can merge it?

**Comment by wd60622** [comment]
> @wd60622 we merged the big CLV refactor PR #298. Do you want to update this one so we can merge it?

Sounds good. I will update it

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Some small bug, as I think the property in the new model already gives you the posterior? Should be easy to fix

**Comment by wd60622** [comment]
Tests adjusted to use the `idata` instead of `fit_results` after the refactor. 
Tests are passing locally

File: pymc_marketing/clv/models/beta_geo.py
Comment (ricardoV94) [inline pymc_marketing/clv/models/beta_geo.py:369]: We should use the property because it will raise the informative error in case the model was not fit yet

```suggestion
            if self.fit_result.dims == {"chain": 1, "draw": 1}:
```

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks @wd60622

**Comment by wd60622** [comment]
> Thanks @wd60622!

Thanks for the feedback

**Pull request merged**
